.github/workflows/ci-pull-requests.yml

```diff
@@ -1,42 +1,54 @@
-name: Test Pull Requests
+name: Build, smoke test and lint branch
 
 on: [pull_request]
 
 jobs:
-  build:
+  e2e-node-16:
+    name: E2E [Electron/Node 16]
+    uses: ./.github/workflows/e2e-reusable.yml
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'skip-e2e') }}
+    with:
+      branch: ${{ github.event.pull_request.base.ref }}
+      user: ${{ github.event.inputs.user || 'PR User' }}
+      spec: ${{ github.event.inputs.spec || 'e2e/0-smoke.cy.ts' }}
+      run-env: base:16.18.1
+      record: false
+      parallel: false
+      containers: '[1]'
+    secrets:
+      CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
+
+  lint:
+    name: Units Tests & Lint
```

Comment:
just an opinion but might be better if unit tests and lints happened before smoke test since they are so much slower.